### ISSUE TITLE
Fix false positives for hex colour containing "e2" for border property in unit-no-unknown

### ISSUE THREAD

**Issue by lobo-tuerto** [stylelint/stylelint#5377]
> Clearly describe the bug

There are false positives when trying to declare a border like this:
```sass
border: 1px solid #5e07e2
```

> Which rule, if any, is the bug related to?

`unit-no-unknown`

> What code is needed to reproduce the bug?

```sass
.some-class
  border: 1px solid #5e07e2
```

> What stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
  extends: ['stylelint-config-standard', 'stylelint-config-prettier'],
  plugins: ['stylelint-order'],
  rules: {
    'at-rule-no-unknown': null,
    'max-nesting-depth': 5,
    'order/properties-alphabetical-order': true,
    'selector-pseudo-element-no-unknown': [
      true,
      {
        ignorePseudoElements: ['v-deep'],
      },
    ],
  },
}

```

> Which version of stylelint are you using?

`13.13.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

In a Nuxt.js project with:

`@nuxtjs/stylelint-module'`

Also, through VSCode `stylelint-vscode` extension.

`stylelint.config.js` is posted above.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Seems related to using SASS syntax (not SCSS).

> What did you expect to happen?

No errors to be raised.

> What actually happened (e.g. what warnings or errors did you get)?

The following errors were raised:

```shell
 ERROR  Failed to compile with 1 errors      friendly-errors
 ERROR  StylelintError      friendly-errors

components/BaseSnackbar.vue
 37:21  ✖  Unexpected unknown unit "e2"   unit-no-unknown
```

It refers to the `e2` at the end of `#5e07e2`.

**Comment by jeddy3** [comment]
@lobo-tuerto Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIB0BnCBbGBaMAbAhiigDoB2ABOQEYQBOst85AjAA4Ae5auAllOQGIArDAAMAdhgAmEABoQKAJ6kALvnYIFhFHPARSAMx4BzTcDKViIWgFdcMEgnLmKlclZukeKrKQhZPAGs-AHdSKyYVWxgLcgBfMjiQOKA).

It's likely a problem with [the parser](https://www.npmjs.com/package/postcss-sass) stylelint uses [to support Sass](https://stylelint.io/about/syntaxes). 

Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes) if you have time as not many people use Sass these days and I believe that package is maintained by one person.

**Comment by jeddy3** [comment]
The [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

Please consider opening an issue upstream and fixing the issue there as the syntax is no longer bundled by Stylelint.
